evm/src/poseidon/columns.rs

```diff
@@ -0,0 +1,94 @@
+use plonky2::field::types::Field;
+
+use crate::cross_table_lookup::Column;
+
+pub(crate) const POSEIDON_SPONGE_WIDTH: usize = 12;
+pub(crate) const POSEIDON_SPONGE_RATE: usize = 8;
+pub(crate) const HALF_N_FULL_ROUNDS: usize = 4;
+pub(crate) const N_PARTIAL_ROUNDS: usize = 22;
+
+/// Registers to hold permutation inputs.
+pub fn reg_input_limb(i: usize) -> usize {
+    debug_assert!(i < POSEIDON_SPONGE_WIDTH);
+    i
+}
+
+pub fn col_input_limb<F: Field>(i: usize) -> Column<F> {
+    Column::single(reg_input_limb(i))
+}
+
+/// Holds x^3 for all elements in full rounds.
+pub fn reg_cubed_full(round: usize, i: usize) -> usize {
+    debug_assert!(i < POSEIDON_SPONGE_WIDTH);
+    debug_assert!(round < 2 * HALF_N_FULL_ROUNDS);
+    POSEIDON_SPONGE_WIDTH + POSEIDON_SPONGE_WIDTH * round + i
+}
+
+const START_POWER_6: usize = POSEIDON_SPONGE_WIDTH + 2 * HALF_N_FULL_ROUNDS * POSEIDON_SPONGE_WIDTH;
+
+/// Holds x^6 for all elements in full rounds.
+pub fn reg_power_6_full(round: usize, i: usize) -> usize {
+    debug_assert!(i < POSEIDON_SPONGE_WIDTH);
+    debug_assert!(round < 2 * HALF_N_FULL_ROUNDS);
+    START_POWER_6 + POSEIDON_SPONGE_WIDTH * round + i
+}
```

Comment:
Random thought here, but as `x^6` is only quadratic with respect to `x^3` which is stored previously, can't we amortize the number of columns here by not storing it directly?